qiskit/opflow/primitive_ops/pauli_op.py

```diff
@@ -314,15 +314,25 @@ def exp_i(self) -> OperatorBase:
             return EvolvedOp(self)
 
     def to_circuit(self) -> QuantumCircuit:
-        # If Pauli equals identity, don't skip the IGates
-        is_identity = sum(self.primitive.x + self.primitive.z) == 0
-
-        # Note: Reversing endianness!!
-        qc = QuantumCircuit(len(self.primitive))
-        for q, pauli_str in enumerate(reversed(self.primitive.to_label())):
-            gate = PAULI_GATE_MAPPING[pauli_str]
-            if not pauli_str == "I" or is_identity:
-                qc.append(gate, qargs=[q])
+
+        pauli = self.primitive.to_label()[-self.num_qubits :]
+        phase = self.primitive.phase
+
+        qc = QuantumCircuit(self.num_qubits)
+        if pauli == "I" * self.num_qubits:
+            qc.global_phase = -phase * pi / 2
+            return qc
+
+        if self.num_qubits == 1:
+            gate = {"I": IGate(), "X": XGate(), "Y": YGate(), "Z": ZGate()}[pauli]
+        else:
+            gate = PauliGate(pauli)
```

Comment:
Is this somehow to optimize things for when num qubits is 1? PauliGate still works right - or it seemed to for me. You think having that special case code in there and the two different paths to maintain is worthwhile

Reply:
Yes. PauliGate speeds up Pauli gates with more than two qubits for Qiskit Aer. For single qubit, it slows down because the transpile time becomes dominant. 

<img width="634" alt="image" src="https://user-images.githubusercontent.com/6814928/121904239-fa50fc80-cd63-11eb-871d-8e566b6e7e59.png">

note: No unitary gates are executed in this example. (all Xgates are canceled)

Reply:
OK, thx for the info.